src/components/menu/menu.test.ts

```diff
@@ -101,3 +101,25 @@ describe('<sl-menu>', () => {
     expect(selectHandler).to.not.have.been.called;
   });
 });
+
+// @see https://github.com/shoelace-style/shoelace/issues/1596
+it('Should fire "sl-select" when clicking an element within a menu-item', async () => {
+  // eslint-disable-next-line
+  const selectHandler = sinon.spy(() => {});
+
+  const menu: SlMenu = await fixture(html`
+    <sl-menu>
+      <sl-menu-label>What Can You Ask Trebellar?</sl-menu-label>
+      <sl-menu-item value="p1">
+        <ui-icon slot="prefix" size="1rem" name="chat"></ui-icon>
+        <span class="q">X</span>
+      </sl-menu-item>
+    </sl-menu>
+  `);
```

Comment:
Can we simplify this to not use the unknown custom element and extra code? Here's a minimal repro if it helps. https://codepen.io/claviska/pen/XWoxamv?editors=1010

Reply:
Done. Sorry I just pulled this directly from the issue.